src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -106,6 +107,7 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool up
         && owner->HasUnitState(UNIT_STATE_FOLLOW));
 
     bool result = i_path->CalculatePath(x, y, z, forceDest);
+    i_path->ReducePathLenghtByDist(owner->GetObjectSize() + i_target->GetObjectSize() + CONTACT_DISTANCE, i_target->ToUnit()); // this distance is correct only for MoveChase with no offset
```

Comment:
there is little point reducing the path length before checking the result in the lines below (also it's length, not lenght)